tensorflow/core/ops/linalg_ops.cc

```diff
@@ -663,4 +663,15 @@ REGISTER_OP("BatchSvd")
     .Deprecated(13, "Use Svd instead.")
     .SetShapeFn(shape_inference::UnknownShape);
 
+#ifdef INTEL_MKL
+
+REGISTER_OP("_MklEinsum")
+    .Input("inputs: N * T")
+    .Output("output: T")
+    .Attr("equation: string")
+    .Attr("N: int >= 1")
+    .Attr("T: {bfloat16, float}")
+    .SetShapeFn(shape_inference::EinsumShape);
+
+#endif  // INTEL_MKL
```

Comment:
Please move this to a separate MKL file, e.g., a new file named `mkl_linalg_ops.cc`. If you don't expect to have more linalg ops in the near future, please feel free to put it in `mkl_nn_ops.cc`.